test/FeatureSetUtils_server_test.py

```diff
@@ -307,6 +307,32 @@ def test_upload_featureset_from_diff_expr(self):
         self.assertTrue('report_name' in result)
         self.assertTrue('report_ref' in result)
 
+        # adding in one test here to make sure that that the filtered expression
+        # matrices each have a proper link in the provenance back to the differential
+        # expression matrix (individual matrix, not set) that was used to create it
+
+        obj = self.wsClient.get_objects([{'ref': self.diff_expression_set_ref}])[0]
+        dl = obj.get('data').get('items')
+        dms = map((lambda r: r.get('ref')),dl)
+
+        # check each filtered expression matrix in the set:
+
+        for fem in result.get('filtered_expression_matrix_ref_list'):
+            prov = self.wsClient.get_object_provenance([{'ref': fem}])[0].get('provenance')[0]
+            self.assertTrue(prov.get( 'input_ws_objects'))
+            dem_list = prov.get('input_ws_objects')
+            self.assertTrue(isinstance(dem_list, list))
+            self.assertTrue(len( dem_list ) == 1)    # should be a list of one
+            self.assertTrue(dem_list[0] in dms)      # and in the diff. expr. set list
```

Comment:
Each of the dem_list should be unique right? Maybe check this to make sure you don't get duplicates?

Reply:
Good idea.  Unfortunately the unit test has duplicates!   (I didn't want to spend time fixing that)

Reply:
Do you think its worth the time and effort?  maybe..

Reply:
Up to you. Mostly want to catch a case where you are returning only one ref but if it's more than one line of code feel free to ignore

Reply:
Actually now that I look again - dem_list should always be a list of exactly one.   Maybe the variable name is misleading

Reply:
Fair enough